src/range.cr

```diff
@@ -312,9 +312,10 @@ struct Range(B, E)
 
     @range : Range(B, E)
     @current : B
-    @reached_end : Bool
+    @reached_end = false
 
-    def initialize(@range : Range(B, E), @current = range.begin, @reached_end = false)
+    def initialize(@range : Range(B, E))
```

Comment:
Here you're also changing the ctor signature, we can no more setup current & reached_end. Is that intentional?

Reply:
It's a private class anyway. And the removed arguments don't seem to be used anywhere. Wouldn't make much sense anyway.

Reply:
The constructor was used internally, so no problem. Same in other places.